{"name": "Смена валюты", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.WebDriverException: Message: unknown error: net::ERR_CONNECTION_REFUSED\n  (Session info: chrome=123.0.6312.105)\nStacktrace:\n#0 0x5ec669131863 <unknown>\n#1 0x5ec668e278c6 <unknown>\n#2 0x5ec668e1f9f3 <unknown>\n#3 0x5ec668e0fb2b <unknown>\n#4 0x5ec668e110c2 <unknown>\n#5 0x5ec668e0fe4c <unknown>\n#6 0x5ec668e0f638 <unknown>\n#7 0x5ec668e0f485 <unknown>\n#8 0x5ec668e0d516 <unknown>\n#9 0x5ec668e0da2a <unknown>\n#10 0x5ec668e2a24e <unknown>\n#11 0x5ec668eb37c5 <unknown>\n#12 0x5ec668e945a2 <unknown>\n#13 0x5ec668eb2c19 <unknown>\n#14 0x5ec668e94343 <unknown>\n#15 0x5ec668e65593 <unknown>\n#16 0x5ec668e65f5e <unknown>\n#17 0x5ec6690f584b <unknown>\n#18 0x5ec6690f97a5 <unknown>\n#19 0x5ec6690e3571 <unknown>\n#20 0x5ec6690fa332 <unknown>\n#21 0x5ec6690c887f <unknown>\n#22 0x5ec669120728 <unknown>\n#23 0x5ec6691208fb <unknown>\n#24 0x5ec6691309b4 <unknown>\n#25 0x745faa894ac3 <unknown>", "trace": "browser = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"71ba8497dfa7e57a0f81f06ee303969d\")>, base_url = 'http://10.0.2.15:8081'\n\n    @allure.title(\"Смена валюты\")\n    def test_change_currency(browser, base_url):\n        header = HeaderElement(browser)\n>       HomePage(browser).open(base_url)\n\ntests/test_start.py:141: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\npage_objects/home_page.py:13: in open\n    self.browser.get(url)\n../../../.local/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:353: in get\n    self.execute(Command.GET, {\"url\": url})\n../../../.local/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:344: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x74f15f9ba620>, response = {'status': 500, 'value': '{\"value\":{\"error\":\"unknown error\",\"message\":\"unknown error: net::ERR_CONNECTION_REFUSED\\\\n  ...\\\\n#23 0x5ec6691208fb \\\\u003Cunknown>\\\\n#24 0x5ec6691309b4 \\\\u003Cunknown>\\\\n#25 0x745faa894ac3 \\\\u003Cunknown>\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.WebDriverException: Message: unknown error: net::ERR_CONNECTION_REFUSED\nE         (Session info: chrome=123.0.6312.105)\nE       Stacktrace:\nE       #0 0x5ec669131863 <unknown>\nE       #1 0x5ec668e278c6 <unknown>\nE       #2 0x5ec668e1f9f3 <unknown>\nE       #3 0x5ec668e0fb2b <unknown>\nE       #4 0x5ec668e110c2 <unknown>\nE       #5 0x5ec668e0fe4c <unknown>\nE       #6 0x5ec668e0f638 <unknown>\nE       #7 0x5ec668e0f485 <unknown>\nE       #8 0x5ec668e0d516 <unknown>\nE       #9 0x5ec668e0da2a <unknown>\nE       #10 0x5ec668e2a24e <unknown>\nE       #11 0x5ec668eb37c5 <unknown>\nE       #12 0x5ec668e945a2 <unknown>\nE       #13 0x5ec668eb2c19 <unknown>\nE       #14 0x5ec668e94343 <unknown>\nE       #15 0x5ec668e65593 <unknown>\nE       #16 0x5ec668e65f5e <unknown>\nE       #17 0x5ec6690f584b <unknown>\nE       #18 0x5ec6690f97a5 <unknown>\nE       #19 0x5ec6690e3571 <unknown>\nE       #20 0x5ec6690fa332 <unknown>\nE       #21 0x5ec6690c887f <unknown>\nE       #22 0x5ec669120728 <unknown>\nE       #23 0x5ec6691208fb <unknown>\nE       #24 0x5ec6691309b4 <unknown>\nE       #25 0x745faa894ac3 <unknown>\n\n../../../.local/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py:229: WebDriverException"}, "steps": [{"name": "Открываю http://10.0.2.15:8081", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.WebDriverException: Message: unknown error: net::ERR_CONNECTION_REFUSED\n  (Session info: chrome=123.0.6312.105)\nStacktrace:\n#0 0x5ec669131863 <unknown>\n#1 0x5ec668e278c6 <unknown>\n#2 0x5ec668e1f9f3 <unknown>\n#3 0x5ec668e0fb2b <unknown>\n#4 0x5ec668e110c2 <unknown>\n#5 0x5ec668e0fe4c <unknown>\n#6 0x5ec668e0f638 <unknown>\n#7 0x5ec668e0f485 <unknown>\n#8 0x5ec668e0d516 <unknown>\n#9 0x5ec668e0da2a <unknown>\n#10 0x5ec668e2a24e <unknown>\n#11 0x5ec668eb37c5 <unknown>\n#12 0x5ec668e945a2 <unknown>\n#13 0x5ec668eb2c19 <unknown>\n#14 0x5ec668e94343 <unknown>\n#15 0x5ec668e65593 <unknown>\n#16 0x5ec668e65f5e <unknown>\n#17 0x5ec6690f584b <unknown>\n#18 0x5ec6690f97a5 <unknown>\n#19 0x5ec6690e3571 <unknown>\n#20 0x5ec6690fa332 <unknown>\n#21 0x5ec6690c887f <unknown>\n#22 0x5ec669120728 <unknown>\n#23 0x5ec6691208fb <unknown>\n#24 0x5ec6691309b4 <unknown>\n#25 0x745faa894ac3 <unknown>\n\n", "trace": "  File \"/home/fuenukkusu2501/Develop/tmp/OTUS_selenium_UI/page_objects/home_page.py\", line 13, in open\n    self.browser.get(url)\n  File \"/home/fuenukkusu2501/.local/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py\", line 353, in get\n    self.execute(Command.GET, {\"url\": url})\n  File \"/home/fuenukkusu2501/.local/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py\", line 344, in execute\n    self.error_handler.check_response(response)\n  File \"/home/fuenukkusu2501/.local/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n"}, "start": 1720461049358, "stop": 1720461050085}], "start": 1720461049354, "stop": 1720461050086, "uuid": "592c0d7e-8bbb-46ab-8826-f0f04578cef3", "historyId": "eec560047c31debe6d9187354a480974", "testCaseId": "eec560047c31debe6d9187354a480974", "fullName": "tests.test_start#test_change_currency", "labels": [{"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_start"}, {"name": "host", "value": "Ubuntu"}, {"name": "thread", "value": "14041-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_start"}]}